app/services/questionnaire_responses/initialize_monthly_supplies_reports.rb

```diff
@@ -0,0 +1,50 @@
+class QuestionnaireResponses::InitializeMonthlySuppliesReports
+  def self.call(date = 1.month.ago)
+    return unless Flipper.enabled?(:monthly_supplies_reports)
+
+    month_date = date.beginning_of_month
+    month_date_str = month_date.strftime("%Y-%m-%d")
+    facility_ids = Facility.select("id").pluck(:id)
+
+    reports_exist = false
+    questionnaire = latest_active_supplies_reports_questionnaire
+
+    facility_ids.each do |facility_id|
+      if monthly_supplies_report_exists?(facility_id, month_date_str)
+        reports_exist = true
+      else
+        QuestionnaireResponse.create!(
+          questionnaire_id: questionnaire.id,
```

Comment:
questionnaire may not exist, we should check for this.

Reply:
done.